File: LeetCode/C++/6. ZigZag Conversion.cpp
///LeetCode 6. ZigZag Conversion by C++
///write by LazyWolfLin in 17.03.05

class Solution {
public:
    string convert(string s, int numRows)
    {
        if (numRows==1) return s;
        string ans;
        int key=2*numRows-2;
        for (int x=0;x<s.size();x+=key) ans+=s[x];
        for (int i=1;i<numRows-1;++i)
        {
            for (int x=i;x<s.size();x+=key)
            {
                ans+=s[x];
                if (x+key-2*i<s.size()) ans+=s[x+key-2*i];
            }
        }
        for (int x=numRows-1;x<s.size();x+=key) ans+=s[x];
        return ans;
    }
};

File: LeetCode/Python3/6. ZigZag Conversion.py
# LeetCode 6. ZigZag Conversion by Python3
# write by LazyWolfLin in 18.08.05

class Solution:
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        key = 2 * numRows - 2
        ans = str()
        for i in range(0, len(s), key):
            ans += s[i]
        for i in range(1, numRows - 1):
            for j in range(i, len(s), key):
                ans += s[j]
                if j + key - 2 * i < len(s):
                    ans += s[j + key - 2 * i]
        for i in range(numRows - 1, len(s), key):
            ans += s[i]
        return ans